argparse_dataclass.py

```diff
@@ -338,6 +341,11 @@ def _add_dataclass_options(
 
         if field.type is bool:
             _handle_bool_type(field, args, kwargs)
+        elif get_origin(field.type) is typing.Union:
+            # Optional[X] is equivalent to Union[X, None].
+            f_args = get_args(field.type)
+            if len(f_args) == 2 and f_args[1] is type(None):
+                kwargs["type"] = f_args[0]
```

Comment:
I guess this is the reason for the above. I'd still feel like this is more future proof if you explicitly figure out which index is `None` and which is the other type. There should probably also be some error checking to explicitly reject other union types since those are not as easy to support.